rust/agama-cli/src/logs.rs

```diff
@@ -45,6 +51,42 @@ pub async fn run(subcommand: LogsCommands) -> anyhow::Result<()> {
     }
 }
 
+// Whatewer passed in dest formed into an absolute path with archive name
+// if dest is none then a default is returned
+// if dest is directory then a default file name for the archive will be appended
+// if dest is path with a file name then it is used as is a name for resulting archive
+fn parse_dest(dest: Option<PathBuf>) -> Result<PathBuf, io::Error> {
+    let default = PathBuf::from(DEFAULT_RESULT);
+    let err = io::Error::new(io::ErrorKind::InvalidInput, "Invalid destination path");
+
+    match dest {
+        None => Ok(default),
+        Some(mut buff) => {
+            let path = buff.as_path();
+
+            // existing directory -> append an archive name
+            if path.is_dir() {
+                buff.push("agama-logs");
+                Ok(buff)
```

Comment:
May I suggest returning early here and simplifying the rest?

```rust
   if path.is_dir() {
       buff.push("agama-logs");
       return Ok(buff)
    }

    if let Some(parent) = path.parent() {
        if parent.is_dir() {
            return Ok(buff)
        }
    }

    Err(err)
```

Reply:
As you wrote, if `is_some_and` would be stable, it might be even simpler. But...

Reply:
may be one day ;-)